special-pages/pages/new-tab/app/widget-list/WidgetList.js

```diff
@@ -36,6 +40,17 @@ export async function widgetEntryPoint(id) {
 
 export function WidgetList() {
     const { widgets, widgetConfigItems, entryPoints } = useContext(WidgetConfigContext);
+    const messaging = useMessaging();
+
+    /**
+     * @param {any} error
+     * @param {{id: string}} widgetConfig
+     */
+    const didCatch = (error, widgetConfig) => {
+        const message = error?.message || error?.error || 'unknown';
+        const composed = `Widget '${widgetConfig.id}' threw an exception: ` + message;
+        messaging.reportPageException({ message: composed });
+    };
```

Comment:
This just gives us more fine-grained information if something starts to fail, and allows the rest of the page to continue.